.github/workflows/ci.yml

```diff
@@ -952,6 +952,66 @@ jobs:
       benchmark-tools-gcs-artifact-dir: ${{ needs.build_benchmark_tools.outputs.benchmark-tools-gcs-artifact-dir }}
       benchmark-presets: ${{ needs.setup.outputs.benchmark-presets }}
 
+  process_benchmark_results:
+    needs: [setup, compilation_benchmarks, execution_benchmarks]
+    # TODO(#11076): Temporarily disabled until we migrate the database.
+    # execution_benchmarks is the optional dependency and skipped in presubmit
+    # if no benchmark is specified to run.
+    if: |
+      false &&
+      always() &&
+      needs.setup.outputs.should-run == 'true' &&
+      needs.setup.outputs.ci-stage == 'presubmit' &&
+      needs.compilation_benchmarks.result == 'success' &&
+      (needs.execution_benchmarks.result == 'success' || needs.execution_benchmarks.result == 'skipped')
+    runs-on:
+      - self-hosted  # must come first
+      - runner-group=${{ needs.setup.outputs.runner-group }}
+      - environment=${{ needs.setup.outputs.runner-env }}
+      - cpu
+      - os-family=Linux
+    env:
+      COMPILE_STATS_RESULTS: ${{ needs.compilation_benchmarks.outputs.compile-stats-results }}
+      COMPILE_STATS_RESULTS_GCS_ARTIFACT: ${{ needs.compilation_benchmarks.outputs.compile-stats-results-gcs-artifact }}
+      EXECUTION_BENCHMARK_RESULTS_DIR: ${{ needs.execution_benchmarks.outputs.benchmark-results-dir }}
+      EXECUTION_BENCHMARK_RESULTS_GCS_ARTIFACT_DIR: ${{ needs.execution_benchmarks.outputs.benchmark-results-gcs-artifact-dir }}
+    steps:
+      - name: "Checking out repository"
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # v2
+      - name: Downloading compilation benchmark results
+        run: |
+          gcloud alpha storage cp \
+            "${COMPILE_STATS_RESULTS_GCS_ARTIFACT}" \
+            "${COMPILE_STATS_RESULTS}"
+      - name: Downloading execution benchmark results
+        id: download-execution-results
+        if: needs.execution_benchmarks.result == 'success'
+        run: |
+          gcloud alpha storage cp -r \
+            "${EXECUTION_BENCHMARK_RESULTS_GCS_ARTIFACT_DIR}" \
+            "${EXECUTION_BENCHMARK_RESULTS_DIR}"
+          echo "execution-benchmark-results-pattern=${EXECUTION_BENCHMARK_RESULTS_DIR}/benchmark-results-*.json" >> "${GITHUB_OUTPUT}"
+      - name: Posting comments
+        env:
+          # Wildcard pattern to match all execution benchmark results. Empty if
+          # execution_benchmarks is skipped, which results in no match.
+          EXECUTION_BENCHMARK_RESULTS_PATTERN: ${{ steps.download-execution-results.outputs.execution-benchmark-results-pattern }}
+          IREE_DASHBOARD_URL: https://perf.iree.dev
+          IREE_BUILD_URL: https://github.com/iree-org/iree/actions/runs/${{ github.run_id }}/attempts/${{ github.run_attempt }}
+        run: |
+          build_tools/github_actions/docker_run.sh \
+            --env "GITHUB_TOKEN=$(gcloud secrets versions access 2 --secret=iree-github-token)" \
```

Comment:
This shouldn't be necessary. The runner gets access to github via its own `GITHUB_TOKEN`. If more access is needed, we should store it in GitHub secrets and should manage this in the GitHub-native way

Reply:
The workflow now is rewritten into two parts. The privileged workflow uses `secrets.WRITE_ACCESS_TOKEN` to post comments.